core/commands/log_graph_colorizer.py

```diff
@@ -0,0 +1,267 @@
+        # type: () -> str
+        return self.char()
+
+    def __repr__(self):
+        return "Char({})".format(self.pt)
+
+    def __hash__(self):
+        # type: () -> int
+        return hash((self.view.id(), self.view.change_count(), self.pt))
+
+    def __eq__(self, rhs):
+        # type: (object) -> bool
+        if isinstance(rhs, str):
+            return self.char() == rhs
+        if isinstance(rhs, self.__class__):
+            return hash(self) == hash(rhs)
+        return NotImplemented
+
+    @property
+    def e(self):
+        # type: () -> Char
+        return self.go((0, 1))
+
+    @property
+    def se(self):
+        # type: () -> Char
+        return self.go((1, 1))
+
+    @property
+    def s(self):
+        # type: () -> Char
+        return self.go((1, 0))
+
+    @property
+    def sw(self):
+        # type: () -> Char
+        return self.go((1, -1))
+
+    @property
+    def w(self):
+        # type: () -> Char
+        return self.go((0, -1))
+
+
+class NullChar_(Char):
+    def __init__(self):
+        # type: () -> None
+        ...
+
+    def __hash__(self):
+        # type: () -> int
+        return hash(None)
+
+    def char(self):
+        # type: () -> str
+        return ' '
+
+    def go(self, rowcol):
+        # type: (RowCol) -> NullChar_
+        return self
+
+
+NullChar = NullChar_()
+registered_handlers = {}  # type: Dict[str, NextFn]
+
+
+# Notes:
+# - We want `follow_char` to be a polymorphic fn. For that we register
+# a handler for each valid graph char using `handles`.
+#
+# - The following implementation for traversing a drawn graph uses
+# `Iterables` (and e.g. `yield from`) to avoid `None` checks everywhere
+# or extensive list concats.
+#
+# - Following a drawing of a graph doesn't follow an algorithm. The
+# code is straight forward, and you basically have to look at some
+# graphs (turn block cursor on in Sublime!) and peek around.
+
+def handles(ch):
+    # type: (str) -> Callable[[NextFn], NextFn]
+    def decorator(fn):
+        # type: (NextFn) -> NextFn
+        if ch in registered_handlers:
+            raise RuntimeError('{} already has a handler registered'.format(ch))
+        registered_handlers[ch] = fn
+        return fn
+
+    return decorator
+
+
+def follow_path(dot):
+    # type: (Char) -> Iterator[Char]
+    for c in follow_char(dot):
+        # print('{} -> {}'.format(dot, c))
+        yield c
+        if c != COMMIT_NODE_CHAR:
+            yield from follow_path(c)
+
+
+def follow_char(char):
+    # type: (Char) -> Iterator[Char]
+    fn = registered_handlers.get(char.char(), follow_none)
+    yield from fn(char)
+
+
+def contains(next_char, test):
+    # type: (Char, str) -> Iterator[Char]
+    if str(next_char) in test:
+        yield next_char
+
+
+@handles(COMMIT_NODE_CHAR)
+def follow_dot(char):
+    # type: (Char) -> Iterator[Char]
+    yield from contains(char.e, '-')
+    yield from contains(char.s, '|' + COMMIT_NODE_CHAR)
+    if char.e != '/':
```

Comment:
Would this match
```
 | 
 ●
/|
```
and not 
```
 | /
 ●/    
/|
```
match this one ?

I have never seen that seen that in a graph

Reply:
Yes, basically on the east is never any char. Iirc correctly I handled '*' and '|' in the same function in the first iteration so this is kind of a left over from that probably. 👍